src/probability/process/stopping.lean

```diff
@@ -1216,4 +1216,66 @@ by { ext ω, rw stopped_value, by_cases hx : ω ∈ s; simp [hx] }
 
 end piecewise_const
 
+section condexp
+/-! ### Conditional expectation with respect to the σ-algebra generated by a stopping time -/
+
+variables [linear_order ι] {μ : measure Ω} {ℱ : filtration ι m} {τ σ : Ω → ι}
+  {E : Type*} [normed_add_comm_group E] [normed_space ℝ E] [complete_space E] {f : Ω → E}
+
+lemma condexp_stopping_time_ae_eq_restrict_eq_of_countable_range [sigma_finite_filtration μ ℱ]
+  (hτ : is_stopping_time ℱ τ) (h_countable : (set.range τ).countable)
+  [sigma_finite (μ.trim (hτ.measurable_space_le_of_countable_range h_countable))] (i : ι) :
+  μ[f | hτ.measurable_space] =ᵐ[μ.restrict {x | τ x = i}] μ[f | ℱ i] :=
+begin
+  refine condexp_ae_eq_restrict_of_measurable_space_eq_on
+    (hτ.measurable_space_le_of_countable_range h_countable) (ℱ.le i)
+    (hτ.measurable_set_eq_of_countable_range' h_countable i) (λ t, _),
+  rw [set.inter_comm _ t, is_stopping_time.measurable_set_inter_eq_iff],
+end
+
+lemma condexp_stopping_time_ae_eq_restrict_eq_of_countable [countable ι]
+  [sigma_finite_filtration μ ℱ]
+  (hτ : is_stopping_time ℱ τ) [sigma_finite (μ.trim hτ.measurable_space_le_of_countable)] (i : ι) :
+  μ[f | hτ.measurable_space] =ᵐ[μ.restrict {x | τ x = i}] μ[f | ℱ i] :=
+condexp_stopping_time_ae_eq_restrict_eq_of_countable_range hτ (set.to_countable _) i
+
+variables [(filter.at_top : filter ι).is_countably_generated]
+
+lemma condexp_min_stopping_time_ae_eq_restrict_le_const
+  (hτ : is_stopping_time ℱ τ) [sigma_finite (μ.trim hτ.measurable_space_le)]
+  [∀ i, sigma_finite (μ.trim (hτ.min_const i).measurable_space_le)] (i : ι) :
```

Comment:
Doesn't the second typeclass assumption imply the first? I mean, if the space is sigma-finite for a small sigma-algebra, then it is also sigma-finite for any larger sigma-algebra as you can use the covering of the space into finite-measure sets given by the first assumption. (Using `sigma_finite_trim_mono` and `measurable_space_mono` or something like that?)